x-pack/plugins/alerting/server/lib/get_next_run_at.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export function getNextRunAt(currentRunAt: Date, interval: number) {
+  let nextRunAt = currentRunAt.getTime() + interval;
+  if (nextRunAt < Date.now()) {
+    // To prevent returning dates in the past, we'll return now instead
+    nextRunAt = Date.now();
```

Comment:
I wonder if what we want to do is respect the original cadence of the interval instead.  Eg, if the alert originally started 0:00 on a 10 min interval, it would only ever fire 6 times per hour, on the tens.  In a past life, I've come to know the "minutes" an alert would go off, and be checked again, kinda useful.

The other aspect of that is that it may help avoid the "thundering herd" if something gets backed up, and then for some reason a bunch of these functions get run at the same time, and so a ton of tasks could get re-scheduled to go off at the same time.

Reply:
The one question I can think of would be what happens if a task takes longer than the interval to execute? (starts at 0:00 but finishes at 0:13 on a 10m interval). Would there be a second task executing at 0:10 while the other one is still running? If I recall correctly from previous discussions, we chosen the path of holding back the next execution until the previous one is finished and the second execution would cover a wider range to "catch up".

Otherwise if that scenario doesn't happen, it would in theory run consistently each 10m but not "on the tens", just whenever it got scheduled + 10m.

In regards to having a bunch of functions running at the same time, the task manager behaves more like a queue in this scenario by configuring the `max_workers`.

Reply:
Just writing down our offline discussion, we've agreed to keep this implementation for now. If ever more complexity / cron like requests come through, we'll see what to do at that time.